File: magellan-sample-migration/src/test/java/com/wealthfront/magellan/sample/migration/TestAppComponent.java
package com.wealthfront.magellan.sample.migration;

import com.wealthfront.magellan.sample.migration.tide.DogDetailsScreenTest;
import com.wealthfront.magellan.sample.migration.tide.DogListStepTest;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = TestAppModule.class)
@Singleton
public interface TestAppComponent extends AppComponent {
  void inject(DogDetailsScreenTest test);

  void inject(DogListStepTest test);
}

File: magellan-sample-migration/src/test/java/com/wealthfront/magellan/sample/migration/TestAppComponent.kt
package com.wealthfront.magellan.sample.migration

import com.wealthfront.magellan.sample.migration.api.DogApi
import com.wealthfront.magellan.sample.migration.tide.DogListStepFactory
import com.wealthfront.magellan.sample.migration.toolbar.ToolbarHelper
import dagger.Component
import javax.inject.Singleton

@Component(modules = [AppModule::class, TestDogApiModule::class, TestToolbarHelperModule::class])
@Singleton
interface TestAppComponent : AppComponent {

  val toolbarHelper: ToolbarHelper
  val api: DogApi
  val dogListStepFactory: DogListStepFactory
}
